pasture-core/src/gpu/device.rs

```diff
@@ -0,0 +1,639 @@
+                    v.extend_from_slice(&one_as_bytes);
+                }
+                return v;
+            }
+            PointAttributeDataType::Vec3u16 => {
+                // Convert to Vec4u32
+                let one_as_bytes = 1_u32.to_ne_bytes();
+
+                // Each entry is 16 bits, ie. 2 bytes -> each Vec3 has 3*2 = 6 bytes
+                let stride = self.bytes_per_element(datatype) as usize;   // = 6
+                let num_elements = len / stride;
+
+                let mut v = Vec::new();
+                for i in 0..num_elements {
+                    // Iteration over each Vec3
+                    for j in 0..3 {
+                        // Extend each entry to 32 bits
+                        let begin = (i * stride) + j * 2;
+                        let end = (i * stride) + (j * 2) + 2;
+
+                        let current = u16::from_ne_bytes(slice[begin..end].try_into().unwrap());
+                        v.extend_from_slice(&(current as u32).to_ne_bytes());
+                    }
+
+                    // Append fourth coordinate
+                    v.extend_from_slice(&one_as_bytes);
+                }
+                return v;
+            }
+            PointAttributeDataType::Vec3f32 => {
+                // Make Vec4f32 by appending 1.0
+                let one_as_bytes = 1.0_f32.to_ne_bytes();
+
+                // Each entry is 64 bits and hence consists of 8 bytes -> a Vec3 has 24 bytes
+                let stride = self.bytes_per_element(datatype) as usize;   // = 24
+                let num_elements = len / stride;
+
+                let mut v: Vec<u8> = Vec::new();
+                for i in 0..num_elements {
+                    let begin = i * stride;
+                    let end = (i * stride) + stride;
+
+                    // Push current Vec3
+                    v.extend_from_slice(&slice[begin..end]);
+
+                    // Push 1 as fourth coordinate
+                    v.extend_from_slice(&one_as_bytes);
+                }
+
+                return v;
+            }
+            PointAttributeDataType::Vec3f64 => {
+                // Make Vec4f64 by appending 1.0
+                let one_as_bytes = 1.0_f64.to_ne_bytes();
+
+                // Each entry is 64 bits and hence consists of 8 bytes -> a Vec3 has 24 bytes
+                let stride = self.bytes_per_element(datatype) as usize;   // = 24
+                let num_elements = len / stride;
+
+                let mut v: Vec<u8> = Vec::new();
+                for i in 0..num_elements {
+                    let begin = i * stride;
+                    let end = (i * stride) + stride;
+
+                    // Push current Vec3
+                    v.extend_from_slice(&slice[begin..end]);
+
+                    // Push 1 as fourth coordinate
+                    v.extend_from_slice(&one_as_bytes);
+                }
+
+                return v;
+            }
+        }
+
+        Vec::from(slice)
```

Comment:
This doesn't feel ideal, as `Vec::from(&[T])` creates a copy of the slice (i.e. allocates new memory). So in case that no alignment is required, this creates an unnecessary copy of the data. An option would be to have a method `needs_alignment` and then call `align_slice` only if the alignment is needed.